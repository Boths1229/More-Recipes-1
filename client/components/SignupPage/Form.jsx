import React from 'react';
import PropTypes from 'prop-types';

const propTypes = {
  handleSubmit: PropTypes.func.isRequired,
  onChange: PropTypes.func.isRequired,
  onFocus: PropTypes.func.isRequired,
  value: PropTypes.shape({
    usernameError: PropTypes.string,
    emailError: PropTypes.string,
    confirmPasswordError: PropTypes.string,
  }),
};

const defaultProps = {
  value: null
};

const Form = props => (
  <div className="container">
    <div className="pt-5" style={{ textAlign: 'center' }}>
      <form className="m-x-auto form card" onSubmit={props.handleSubmit}>
        <h1 className="signin-text">More-Recipes</h1>
        <div className="pb-4 pt-3 form-group">
          <input
            type="text"
            name="username"
            className="form-control"
            required
            onChange={props.onChange}
            onFocus={props.onFocus}
            placeholder="Username"
          />
          {
            props.value.usernameError &&
            <div className="alert alert-danger">
              {
                props.value.usernameError
              }
            </div>
          }
        </div>
        <div className="pb-4 form-group">
          <input
            type="email"
            name="email"
            className="form-control"
            required
            onFocus={props.onFocus}
            onChange={props.onChange}
            placeholder="Email"
          />
          {
            props.value.emailError &&
            <div className="alert alert-danger">
              {
                props.value.emailError
              }
            </div>
          }
        </div>
        <div className="pb-4 form-group">
          <input
            type="password"
            id="pass"
            name="password"
            pattern=".{7,12}"
            title="7 to 12 characters"
            className="form-control"
            required
            onFocus={props.onFocus}
            onChange={props.onChange}
            placeholder="Password"
          />
        </div>
        <div className="pb-4 form-group">
          <input
            type="password"
            name="confirmPassword"
            className="form-control"
            required
            onFocus={props.onFocus}
            onChange={props.onChange}
            placeholder="Confirm Password"
          />
          {
            props.value.confirmPasswordError &&
            <div className="alert alert-danger">
              {
                props.value.confirmPasswordError
              }
            </div>
          }
        </div>
        <button type="submit" className="btn signin-btn btn-lg" style={{ color: '#f2b43c' }}>
          Create Account
        </button>
        <div className="form-group">
          <p>
            Already have an account? <a href="/signin">Log in.</a>
          </p>
        </div>
      </form>
    </div>
  </div>
);
Form.propTypes = propTypes;
Form.defaultProps = defaultProps;

export default Form;
