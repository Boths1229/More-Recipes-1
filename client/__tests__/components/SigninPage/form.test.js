import React from 'react';
import { shallow } from 'enzyme';
import Form from '../../../components/SigninPage/Form';

let props;

const setup = () => {
  props = {
    error: {},
    onChange: () => {},
    onSubmit: () => {},
    onFocus: () => {},
  };

  return shallow(<Form {...props} />);
};

describe('Signin Form Component', () => {
  it('should render correctly', () => {
    const wrapper = setup();
    expect(wrapper.find('form').length).toBe(1);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render error if error exist', () => {
    const wrapper = setup();
    wrapper.setProps({ error: { error: 'User not found' } });
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('div.alert').at(0).text()).toBe('User not found');
  });
});
