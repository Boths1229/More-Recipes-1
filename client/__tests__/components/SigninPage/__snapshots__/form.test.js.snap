// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signin Form Component should render correctly 1`] = `
<div
  className="container"
>
  <div
    className="pt-5"
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    <form
      className="m-x-auto form card"
      onSubmit={[Function]}
    >
      <h1
        className="signin-text"
      >
         More-Recipes
      </h1>
      <div>
        <img
          alt="logo"
          className="img-fluid rounded-circle img-size"
          name="image"
          src="test"
        />
      </div>
      <div
        className="pb-4 pt-3 form-group"
      >
        <input
          className="form-control"
          name="username"
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="Username or Email"
          required={true}
          type="text"
        />
      </div>
      <div
        className="pb-4 form-group"
      >
        <input
          className="form-control"
          name="password"
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="Password"
          required={true}
          type="password"
        />
      </div>
      <button
        className="btn signin-btn btn-lg"
        style={
          Object {
            "color": "#f2b43c",
          }
        }
        type="submit"
      >
        Log in
      </button>
      <p>
         Don’t have an account?
        <a
          href="/signup"
        >
          Create an account.
        </a>
      </p>
    </form>
  </div>
</div>
`;

exports[`Signin Form Component should render error if error exist 1`] = `
<div
  className="container"
>
  <div
    className="pt-5"
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    <form
      className="m-x-auto form card"
      onSubmit={[Function]}
    >
      <h1
        className="signin-text"
      >
         More-Recipes
      </h1>
      <div>
        <img
          alt="logo"
          className="img-fluid rounded-circle img-size"
          name="image"
          src="test"
        />
      </div>
      <div
        className="alert alert-danger"
      >
        User not found
      </div>
      <div
        className="pb-4 pt-3 form-group"
      >
        <input
          className="form-control"
          name="username"
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="Username or Email"
          required={true}
          type="text"
        />
      </div>
      <div
        className="pb-4 form-group"
      >
        <input
          className="form-control"
          name="password"
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="Password"
          required={true}
          type="password"
        />
      </div>
      <button
        className="btn signin-btn btn-lg"
        style={
          Object {
            "color": "#f2b43c",
          }
        }
        type="submit"
      >
        Log in
      </button>
      <p>
         Don’t have an account?
        <a
          href="/signup"
        >
          Create an account.
        </a>
      </p>
    </form>
  </div>
</div>
`;
